src/others/components/FilterDropdown/ListWithSearch.tsx

```diff
@@ -0,0 +1,113 @@
+import { FixedSizeList, areEqual } from "react-window";
+import { Button, ListItem, ListItemButton, ListItemText, ClickAwayListener, Divider, useAutocomplete, Typography } from "@mui/material";
+import { FunctionComponent, memo, forwardRef } from "react";
+import SearchIcon from "@mui/icons-material/Search";
+
+import { FilterItem } from "../../contexts/filter";
+
+import styles from "./ListWithSearch.module.css";
+
+type ListWithSearchProps = {
+  selectedFilterItemCount: number;
+  selectedFilterItems: FilterItem[];
+  searchableItems: FilterItem[];
+  onSelectItem: (id: string, selected: boolean) => () => void;
+  checkboxListItemIcon: (value: boolean) => React.ReactNode;
+  toggleFilterList: () => void;
+  clearAllFilters: () => void;
+};
+
+function itemKey(index: number, data: FilterItem[]) {
+  const item = data[index];
+  return `${item.id}-${index}`;
+}
+
+export const ListWithSearch: FunctionComponent<ListWithSearchProps> = ({
+  searchableItems,
+  selectedFilterItemCount,
+  selectedFilterItems,
+  onSelectItem,
+  checkboxListItemIcon,
+  toggleFilterList,
+  clearAllFilters,
+}) => {
+  const { getRootProps, getInputProps, getListboxProps, groupedOptions } = useAutocomplete<FilterItem>({
+    id: "search_filters_list",
+    options: searchableItems,
+    getOptionLabel: (option) => option.text,
+    open: true,
+  });
+
+  const Row = memo((props: { data: FilterItem[]; index: number; style: object }) => {
+    const { data, index, style } = props;
+    const { id, selected, text } = data[index];
+
+    return (
+      <ListItem key={id} style={style}>
+        <ListItemButton onClick={onSelectItem(id, selected)} sx={{ paddingRight: "8px" }}>
+          <ListItemText>{text}</ListItemText>
+          {checkboxListItemIcon(selected)}
+        </ListItemButton>
+      </ListItem>
+    );
+  }, areEqual);
+
+  const InnerElementType = forwardRef<HTMLUListElement>((props, ref) => {
+    return <ul ref={ref} {...props} {...getListboxProps()} />;
+  });
+
+  return (
+    <ClickAwayListener onClickAway={toggleFilterList}>
+      <div className={styles.listContainer}>
+        <div className={styles.clearButtonContainer}>
+          <Button variant="text" size="large" onClick={clearAllFilters}>
+            Clear
+          </Button>
+        </div>
+
+        <div {...getRootProps()} className={styles.searchContainer}>
+          <SearchIcon sx={{ color: "000" }} />
+          <input {...getInputProps()} placeholder="Search your city" className={styles.searchInput} />
```

Comment:
**question**

Dont we want to use MUI's Text Field?

Reply:
I think this is correct, as we are passing the `getInputProps()` from the `useAutocomplete` hook, which I don't think is compatible with `TextField`.

Example here - https://mui.com/components/autocomplete/#useautocomplete